### PR TITLE
Add user's profile page.

### PR DESCRIPTION
**Pull request by MFJess**
Why:

* We need to have a page where users can see and update their personal information, and also their answers to previous weekly questions.

This change addresses the need by:

* Fetch information of the current user from the back-end and display name, email, profile picture, discord, company, office, hobbies and skills.
* Fetch all the answers assigned to the specified user, as well as the corresponding questions.

### CODE REVIEW

File: src/components/list.js [hunk 1]
```diff
@@ -0,0 +1,15 @@
+const List = ({ things }) => {
```
Comment (jopedroliveira) [inline src/components/list.js:1]: "thing" is not the best name for an argument. No need to refactor, but keep in mind naming is very important

File: src/components/list.js [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+  return (
+    <div className="mt-6 flex space-x-4">
+      {things.map((thing, index) => (
+        <div key={index}>
+          <div className="bg-lilac text-white text-base inline-block py-2 px-4 mt-4 border-2 border-white rounded-[20px]">
```
Comment (jopedroliveira) [inline src/components/list.js:6]: Although this is not necessarily wrong, there's no need to create separated divs here.

File: src/pages/friend.js [hunk 3]
```diff
@@ -17,7 +17,7 @@ import NavBar from "../components/navBar";
 import { GET } from "../utils/fetch";
 import { useState } from "react";
 
-const current_user_id = localStorage.getItem("current");
+const current_user_id = 1; //localStorage.getItem("current");
```
Comment (jopedroliveira) [inline src/pages/friend.js:20]: Was this change intentional?  Will `current_user_id` always be 1?

File: src/pages/friend.js [hunk 4]
```diff
@@ -39,166 +39,154 @@ function Friend() {
...
+            <Link to="https://skribbl.io">
+              <button
+                className="w-72 h-20 relative flex flex-col items-center justify-center w-72 h-20 rounded-[27px] text-white text-lg font-bold bg-cover bg-center focus:border-4 focus:border-white"
+                style={{ backgroundImage: `url(${Skriblio})` }}
+                onMouseOver={(e) =>
+                  (e.target.innerText =
+                    "Draw and guess what your friends are drawing")
+                }
+                onMouseLeave={(e) => (e.target.innerText = "Skribl.io")}
+              >
```
Comment (jopedroliveira) [inline src/pages/friend.js:118]: Mixing logic handling on markup is not a good practice. Is not completely wrong tho.
The correct approach would be creating a function on the component ex `mouseLeaveHandler` and referring to it on the markup. That way, markup will just be markup, and logic will be contained

File: src/pages/friend.js [hunk 5]
```diff
@@ -39,166 +39,154 @@ function Friend() {
...
+            </Link>
+
+            <Link to="https://playingcards.io/">
+              <button
+                class="w-72 h-20 relative flex flex-col items-center justify-center w-72 h-20 rounded-[27px] text-white text-lg font-bold bg-cover bg-center focus:border-4 focus:border-white"
+                style={{ backgroundImage: `url(${CardGames})` }}
+                onMouseOver={(e) =>
+                  (e.target.innerText = "Play from a selection of card games!")
+                }
+                onMouseLeave={(e) => (e.target.innerText = "Cards Games")}
```
Comment (jopedroliveira) [inline src/pages/friend.js:130]: same as previous

File: src/pages/friend.js [hunk 6]
```diff
@@ -39,166 +39,154 @@ function Friend() {
...
+
+            <Link to="http://en.battleship-game.org/">
+              <button
+                class="w-72 h-20 relative flex flex-col items-center justify-center w-72 h-20 rounded-[27px] text-white text-lg font-bold bg-cover bg-center focus:border-4 focus:border-white"
+                style={{ backgroundImage: `url(${Battleship})` }}
+                onMouseOver={(e) =>
+                  (e.target.innerText =
+                    "Play the online version of the classic battleship game")
+                }
+                onMouseLeave={(e) => (e.target.innerText = "Battleship")}
```
Comment (jopedroliveira) [inline src/pages/friend.js:144]: same as previous

File: src/pages/friend.js [hunk 7]
```diff
@@ -39,166 +39,154 @@ function Friend() {
...
+            <Link to="https://toughlovearena.com/">
+              <button
+                class="w-72 h-20 relative flex flex-col items-center justify-center w-72 h-20 rounded-[27px] text-white text-lg font-bold bg-cover bg-center focus:border-4 focus:border-white"
+                style={{ backgroundImage: `url(${OnlineMiniGames})` }}
+                onMouseOver={(e) =>
+                  (e.target.innerText =
+                    "Play a fun mini-game with your friend!")
+                }
+                onMouseLeave={(e) => (e.target.innerText = "Online mini-games")}
+              >
```
Comment (jopedroliveira) [inline src/pages/friend.js:159]: same as previous

File: src/pages/friend.js [hunk 8]
```diff
@@ -39,166 +39,154 @@ function Friend() {
...
+            <Link to="https://www.chess.com">
+              <button
+                class="w-72 h-20 relative flex flex-col items-center justify-center w-72 h-20 rounded-[27px] text-white text-lg font-bold bg-cover bg-center focus:border-4 focus:border-white"
+                style={{ backgroundImage: `url(${Chess})` }}
+                onMouseOver={(e) =>
+                  (e.target.innerText =
+                    "Play the classic chess game with your friend")
+                }
+                onMouseLeave={(e) => (e.target.innerText = "Chess")}
+              >
```
Comment (jopedroliveira) [inline src/pages/friend.js:173]: same as previous

File: src/pages/friend.js [hunk 9]
```diff
@@ -39,166 +39,154 @@ function Friend() {
...
+              </button>
+            </Link>
+
+            <Link to="https://www.trivianerd.com/">
+              <button
+                class="w-72 h-20 relative flex flex-col items-center justify-center w-72 h-20 rounded-[27px] text-white text-lg font-bold bg-cover bg-center focus:border-4 focus:border-white"
+                style={{ backgroundImage: `url(${TriviaGames})` }}
+                onMouseOver={(e) =>
+                  (e.target.innerText = "Answer questions about any topic!")
+                }
```
Comment (jopedroliveira) [inline src/pages/friend.js:184]: same as previous

File: src/pages/profile.js [hunk 10]
```diff
@@ -0,0 +1,177 @@
...
+};
+
+const getUserData = async () => {
+  var userData = await GET("users/" + current_user_id);
+  var company = await GET("/companies/" + userData.company_id.toString());
+  return {
+    userData: userData,
+    company: company,
+  };
+};
```
Comment (jopedroliveira) [inline src/pages/profile.js:57]: This logic should be isolated. You can create a custom hook and have all the requests there, then when a component needs it, you just need call from hook

File: src/pages/profile.js [hunk 11]
```diff
@@ -0,0 +1,177 @@
...
+
+const formatDate = (dateString) => {
+  const date = new Date(dateString);
+  const options = { month: "short", day: "2-digit" };
+  return date.toLocaleString("en-US", options);
+};
+
+const getAnswers = async () => {
+  var response = await GET("users/" + current_user_id + "/answers");
```
Comment (jopedroliveira) [inline src/pages/profile.js:36]: [Template literals](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals) might be useful for dynamic  strings

File: src/pages/friend.js (hunk 3 above)
Comment (MFJess) [reply src/pages/friend.js:20]: That wasn't suppposed to be that way. I dont know why but this PR has changes in files it shouldn't have. I'm only working in `profile.js` in this one (and related components `list.js` and `Info`
The up to date changes in `friend.js` are in main, I just checked. I need to solve conflicts apparently

**Review by jopedroliveira — APPROVED** [review]

**Comment by jopedroliveira** [comment]
Overall looks good 🚀

**Review by jopedroliveira — APPROVED** [review]

File: src/pages/profile.js [hunk 12]
```diff
@@ -0,0 +1,149 @@
...
+  const options = { month: "short", day: "2-digit" };
+  return date.toLocaleString("en-US", options);
+};
+
+function Profile() {
+  const [answers, setAnswers] = useState([]);
+  const [userData, setUserData] = useState([]);
+  const [skills, setSkills] = useState([]);
+  const [hobbies, setHobbies] = useState([]);
+  const [company, setCompany] = useState([]);
```
Comment (jopedroliveira) [inline src/pages/profile.js:46]: Is this supposed to be an array by default?

Comment (MFJess) [reply src/pages/profile.js:46]: yes because I have to map those

File: src/utils/getters.js [hunk 13]
```diff
@@ -0,0 +1,26 @@
+import { GET } from "../utils/fetch";
+
+export async function getAnswers(user_id) {
+  return await GET(`users/${user_id}/answers`);
+}
+
+export async function getSkills(user_id) {
+  var response = await GET(`users/${user_id}/skills`);
+  console.log("Skills API Response:", response);
```
[the remark this answers is not in the document]
Comment (MFJess) [reply src/utils/getters.js:9]: true! thanks
